charts/argo-controller/templates/config.yaml

```diff
@@ -49,5 +49,11 @@ data:
       secondsAfterSuccess: {{ .workflowTTL.secondsAfterSuccess }}
       secondsAfterFailure: {{ .workflowTTL.secondsAfterFailure }}
     {{- end }}
+  {{- with .Values.prometheusMetrics }}
+  metricsConfig: |
+    disableLegacy: true
+    enabled: {{ .enabled }}
+    path: {{ .path }}
+    port: {{ .port }}
```

Comment:
One final complication: We need to make sure that this port is exposed outside of the container. The setting needs to go in the [deployment](https://github.com/broadinstitute/monster-helm/blob/master/charts/argo-controller/templates/deployment.yaml) as part of the container spec. For reference, [here's](https://github.com/broadinstitute/monster-helm/blob/master/charts/argo-server/templates/deployment.yaml#L33-L35) how the argo-server chart exposes its port.